app/services/AuditService.scala

```diff
@@ -23,24 +23,30 @@ import play.api.libs.json.JsValue
 import play.api.mvc.AnyContent
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.audit.http.connector.AuditConnector
-
 import javax.inject.Inject
+import models.FileDetails
+
 import scala.concurrent.ExecutionContext.Implicits._
 
 class AuditService @Inject()(auditConnector: AuditConnector,
                              localDateTimeService: LocalDateTimeService) {
 
-  def audit(event: String, response: Option[JsValue] = None)(implicit request: IdentifierRequest[AnyContent], hc: HeaderCarrier): Unit = {
+  def audit(event: String,
+            response: Option[JsValue] = None,
+            fileDetails: Option[FileDetails] = None
+           )(implicit request: IdentifierRequest[AnyContent], hc: HeaderCarrier): Unit = {
 
     val payload = ObligedEntityAuditEvent(
       internalAuthId = request.internalId,
       identifier = request.identifier.value,
       affinity = request.affinityGroup,
       dateTime = request.headers.get(DATE).getOrElse(localDateTimeService.now.toString),
-      response = response
+      response = response,
```

Comment:
We will end up with keys here that will have null values if fileDetails is None?

Reply:
I've split up the AuditEvents to remove the None's, and added the fileGenerationDateTime into the AuditFileDetailsEvent one